File: backend/app/models/property.py
from sqlalchemy import Column, Integer, String, Float, DateTime, ForeignKey
from sqlalchemy.orm import relationship
from datetime import datetime, timezone
from app.database import Base

class Property(Base):
    __tablename__ = "properties"

    id = Column(Integer, primary_key=True, index=True)
    address = Column(String, index=True)
    city = Column(String, index=True)
    state = Column(String, index=True)
    zip_code = Column(String, index=True)
    property_type = Column(String, index=True)
    bedrooms = Column(Float)
    bathrooms = Column(Float)
    square_feet = Column(Integer)
    lot_size = Column(Float)
    year_built = Column(Integer)
    current_value = Column(Float)
    purchase_price = Column(Float)
    created_at = Column(DateTime, default=datetime.now(timezone.utc))
    updated_at = Column(DateTime, default=datetime.now(timezone.utc), onupdate=datetime.now(timezone.utc))

    # Relationships
    sales = relationship("Sale", back_populates="property")
    renovations = relationship("Renovation", back_populates="property")

    def __repr__(self):
        return f"<Property {self.address}, {self.city}, {self.state}>" 

File: frontend/src/types/property.ts
export interface Property {
  id: number;
  address: string;
  city: string;
  state: string;
  zip_code: string;
  property_type: string;
  bedrooms: number;
  bathrooms: number;
  square_feet: number;
  lot_size: number;
  year_built: number;
  current_value: number;
  purchase_price: number;
  purchase_date: string;
  last_sale_price: number;
  last_sale_date: string;
  created_at: string;
  updated_at: string;
}
